wear/src/main/java/io/homeassistant/companion/android/home/views/SettingsView.kt

```diff
@@ -1,66 +1,175 @@
-                Image(asset = CommunityMaterial.Icon.cmd_delete)
-            },
-            label = {
-                Text(
-                    text = stringResource(id = R.string.clear_favorites),
+        }
+
+        item {
+            Chip(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(top = 10.dp),
+                icon = {
+                    Image(asset = CommunityMaterial.Icon.cmd_delete)
+                },
+                label = {
+                    Text(
+                        text = stringResource(id = R.string.clear_favorites),
+                    )
+                },
+                onClick = onClearFavorites,
+                colors = ChipDefaults.primaryChipColors(
+                    contentColor = Color.Black
+                ),
+                secondaryLabel = {
+                    Text(
+                        text = stringResource(id = R.string.irreverisble)
+                    )
+                },
+                enabled = favorites.isNotEmpty()
+            )
+        }
+
+        item {
+            val haptic = LocalHapticFeedback.current
+            var checked by rememberSaveable { mutableStateOf(isHapticEnabled) }
+            ToggleChip(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(top = 10.dp),
+                checked = checked,
+                onCheckedChange = {
+                    checked = it
+                    onHapticEnabled(it)
+                    haptic.performHapticFeedback(HapticFeedbackType.LongPress)
+                },
+                label = {
+                    Text(stringResource(R.string.setting_haptic_label))
+                },
+                appIcon = {
+                    Image(
+                        asset =
+                        if (checked)
+                            CommunityMaterial.Icon3.cmd_watch_vibrate
+                        else
+                            CommunityMaterial.Icon3.cmd_watch_vibrate_off
+                    )
+                },
+                colors = ToggleChipDefaults.toggleChipColors(
+                    checkedStartBackgroundColor = Color(0xFFAECBFA),
+                    checkedEndBackgroundColor = Color(0xFFAECBFA),
+                    uncheckedStartBackgroundColor = Color(0xFFAECBFA),
+                    uncheckedEndBackgroundColor = Color(0xFFAECBFA),
+                    checkedContentColor = Color.Black,
+                    uncheckedContentColor = Color.Black,
                 )
-            },
-            onClick = onClearFavorites,
-            colors = ChipDefaults.primaryChipColors(
-                contentColor = Color.Black
-            ),
-            secondaryLabel = {
-                Text(
-                    text = stringResource(id = R.string.irreverisble)
+            )
+        }
+        item {
+            var checked by rememberSaveable { mutableStateOf(isToastEnabled) }
```

Comment:
We don't want to remember between recompositions, we should always use `isToastEnabled`

Reply:
so without this and even with using the viewmodel variable the checked status does not update until the page is recomposed which only happens when I navigate to another screen and back